app/lib/pathutils.js

```diff
@@ -1,7 +1,18 @@
 import path from "path"
 import dateFormat from "dateformat"
+import fs from "fs-extra"
 
-export const getClonePath = (basePath, assignmentName, studentUsername) => {
+export const getAssignmentFolder = (basePath, assignmentName) => {
   const time = dateFormat(new Date(), "mm-dd-yyyy-hh-MM-ss")
-  return path.join(basePath, `${assignmentName}-${time}`, studentUsername)
+  return path.join(basePath, `${assignmentName}-${time}`)
+}
+
+export const getClonePath = async (basePath, studentUsername) => {
+  const submissionPath = path.join(basePath, studentUsername)
+  try {
+    await fs.ensureDir(submissionPath)
+    return submissionPath
+  } catch (error) {
+    return null
```

Comment:
What happens when trap this error?   It looks like in the submissionViewDirectory function we try and open the path if it is not equal to an empty string. I think the path would be `null` in this scenario

```      const submission = candidates[0]
      if (submission.clonePath !== "") {
        opener(submission.clonePath)
      }

```

Reply:
Just added a check for an error from this method in `submission-clone`, if this fails I display `Clone failed: Folder could not be created.` for that submission